include/envoy/upstream/health_checker.h

```diff
@@ -10,7 +10,7 @@ namespace Upstream {
 
 enum class HealthState { Unhealthy, Healthy };
 
-enum class HealthTransition { Unchanged, Changed };
+enum class HealthTransition { Unchanged, Changed, ChangePending };
```

Comment:
Can you add some detailed comments on what each of these mean for future readers? Also, a general comment overview of the flow and how we pick the timeouts I think would be super helpful.

Reply:
Of course.